pkg/action/upgrade.go

```diff
@@ -233,6 +233,13 @@ func (u *Upgrade) prepareUpgrade(name string, chart *chart.Chart, vals map[strin
 func (u *Upgrade) performUpgrade(originalRelease, upgradedRelease *release.Release) (*release.Release, error) {
 	current, err := u.cfg.KubeClient.Build(bytes.NewBufferString(originalRelease.Manifest), false)
 	if err != nil {
+		// Checking for Kubernetes deprecated APIs error so can provide a more informative error message to the user
+		// Ref: https://github.com/helm/helm/issues/7219
+		if strings.Contains(err.Error(), "unable to recognize \"\": no matches for kind") {
+			return upgradedRelease, errors.Wrap(err, "current release manifest contains deprecated kubernetes api(s) for this "+
```

Comment:
I don't mean to be a stickler on terminology but IIUC a "deprecated" API won't cause a problem but a "removed" API (which was deprecated in the previous versions of kube) will cause the problem.

So the work message should probably replace "deprecated" with "removed".

Reply:
Good point. Updated.